fileshare/libdrop/libdrop.go

```diff
@@ -2,114 +2,247 @@
 package libdrop
 
 import (
-	"encoding/json"
 	"fmt"
 	"log"
 	"net/netip"
+	"path/filepath"
+	"strings"
 	"sync"
 	"time"
 
-	norddropgo "github.com/NordSecurity/libdrop/norddrop/ffi/bindings/linux/go"
+	norddrop "github.com/NordSecurity/libdrop-go/v7"
 	"github.com/NordSecurity/nordvpn-linux/fileshare"
+	"github.com/NordSecurity/nordvpn-linux/fileshare/pb"
 	"github.com/NordSecurity/nordvpn-linux/internal"
+	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
 // Fileshare is the main functional filesharing implementation using norddrop library.
 // Thread safe.
 type Fileshare struct {
-	norddrop     norddropgo.Norddrop
+	norddrop     *norddrop.NordDrop
 	eventsDbPath string
-	appVersion   string
 	storagePath  string
 	isProd       bool
 	mutex        sync.Mutex
 }
 
-func logLevelToPrefix(level norddropgo.Enum_SS_norddrop_log_level) string {
+func logLevelToPrefix(level norddrop.LogLevel) string {
 	switch level {
-	case norddropgo.NORDDROPLOGCRITICAL, norddropgo.NORDDROPLOGERROR:
+	case norddrop.LogLevelCritical, norddrop.LogLevelError:
 		return internal.ErrorPrefix
-	case norddropgo.NORDDROPLOGWARNING:
+	case norddrop.LogLevelWarning:
 		return internal.WarningPrefix
-	case norddropgo.NORDDROPLOGDEBUG, norddropgo.NORDDROPLOGTRACE:
+	case norddrop.LogLevelDebug, norddrop.LogLevelTrace:
 		return internal.DebugPrefix
+	case norddrop.LogLevelInfo:
+		return internal.InfoPrefix
 	default:
 		return internal.InfoPrefix
 	}
 }
 
-func logCB(level int, message string) {
-	log.Println(
-		logLevelToPrefix(norddropgo.Enum_SS_norddrop_log_level(level)),
-		"DROP("+norddropgo.NorddropVersion()+"): "+message,
-	)
+type defaultKeyStore struct {
+	pubkeyFunc func(string) []byte
+	privKey    string
+}
+
+func (dks defaultKeyStore) OnPubkey(peer string) *[]byte {
+	pubKey := dks.pubkeyFunc(peer)
+	return &pubKey
+}
+
+func (dks defaultKeyStore) Privkey() []byte {
+	return []byte(dks.privKey)
+}
+
+type defaultLogger struct {
+	logLevel norddrop.LogLevel
+}
+
+func (dl defaultLogger) OnLog(level norddrop.LogLevel, msg string) {
+	log.Println(logLevelToPrefix(level), "DROP("+norddrop.Version()+"): "+msg)
+}
+
+func (dl defaultLogger) Level() norddrop.LogLevel {
+	return dl.logLevel
+}
+
+type libdropEventCallback struct {
+	eventCallback fileshare.EventCallback
+}
+
+func (nec libdropEventCallback) OnEvent(nev norddrop.Event) {
+	ev := libdropEventToInternalEvent(nev)
+	nec.eventCallback.OnEvent(ev)
+}
+
+func libdropEventToInternalEvent(nev norddrop.Event) fileshare.Event {
+	return fileshare.Event{
+		Kind:      toInternalEventKind(nev.Kind),
+		Timestamp: nev.Timestamp,
+	}
+}
+
+func toInternalEventKind(kind norddrop.EventKind) fileshare.EventKind {
```

Comment:
maybe now is too late, but lots of the structs from here are almost identical with the ones from `norddrop`.
does it make sense to have this conversion, wouldn't be easier to have just an alias for the `norddrop` types?

Reply:
The issue is that if I introduce the aliases for those types, I need to have a reference to `norddrop` types in those aliases, like:
```
type EventKindRequestReceived = norddrop.EventKindRequestReceived
```
This will create circular dependency issue, which can be dealt with, but most importantly, this will pollute the fileshare package with implementation details. Right now, I'm using `libdrop` package only in `fileshare/main` and injecting it where needed and fileshare has no idea about libdrop details